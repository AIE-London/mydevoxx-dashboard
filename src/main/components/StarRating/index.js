/**
 * Created by dan on 04/04/2017.
 */
import React, { Component } from 'react';
import styled from 'styled-components';
import {Row} from 'react-flexbox-grid';

const Star = styled.div`
  margin: 6.25px 0;
  position: relative;
  display: block;
  color: red;
  width: 0px;
  opacity: 0.3;
  height: 0px;
  border-right:  12.5px solid transparent;
  border-bottom: 8.75px  solid grey;
  border-left:   12.5px solid transparent;
  -moz-transform:    rotate(35deg);
  -webkit-transform: rotate(35deg);
  -ms-transform:     rotate(35deg);
  -o-transform:      rotate(35deg);
  &:before {
    border-bottom: 10px solid grey;
    border-left: 3.75px solid transparent;
    border-right: 3.75px solid transparent;
    position: absolute;
    height: 0;
    width: 0;
    top: -5.625px;
    left: -8.125px;
    display: block;
    content: '';
    -webkit-transform: rotate(-35deg);
    -moz-transform:    rotate(-35deg);
    -ms-transform:     rotate(-35deg);
    -o-transform:      rotate(-35deg);
  }
  &:after {
    position: absolute;
    display: block;
    color: red;
    top: 1px;
    left: -13px;
    width: 0px;
    height: 0px;
    border-right: 12.5px solid transparent;
    border-bottom: 8.75px solid grey;
    border-left: 12.5px solid transparent;
    -webkit-transform: rotate(-70deg);
    -moz-transform:    rotate(-70deg);
    -ms-transform:     rotate(-70deg);
    -o-transform:      rotate(-70deg);
    content: '';
  }
`;
const ActiveStar = styled(Star)`
  opacity: 1;
  border-bottom-color: #ff9e19;
  &:before {
    border-bottom-color: #ff9e19;
  }
  &:after {
    border-bottom-color: #ff9e19;
  }
`;
class StarRating extends Component {
  getStars(rating){
    var stars = [];
    for(var i = 1; i <= 5; i++) {
      if (i <= rating){
        stars.push(<ActiveStar/>);
      } else {
        stars.push(<Star/>);
      }
    }
    return stars;
  }
  render() {
    return (
      <Row left="xs">
        {this.getStars(this.props.rating)}
      </Row>
    );
  }
}

export default StarRating;
