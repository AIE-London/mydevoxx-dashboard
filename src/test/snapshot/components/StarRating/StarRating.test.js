/**
 * Created by dan on 04/04/2017.
 */
import React from 'react';
import ReactDOM from 'react-dom';
import StarRating from '../../../../main/components/StarRating';
import renderer from 'react-test-renderer';

test('StarRating', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<StarRating />, div);
  });

});
test('StarRating with rating 0', () => {

  const tree = renderer.create(
    <StarRating rating="0" />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});
test('StarRating with rating 5', () => {

  const tree = renderer.create(
    <StarRating rating="5" />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});
test('StarRating with rating 4', () => {

  const tree = renderer.create(
    <StarRating rating="4" />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});
test('StarRating with rating 3', () => {

  const tree = renderer.create(
    <StarRating rating="3" />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});
test('StarRating with rating 2', () => {

  const tree = renderer.create(
    <StarRating rating="2" />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});
test('StarRating with rating 1', () => {

  const tree = renderer.create(
    <StarRating rating="1" />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});