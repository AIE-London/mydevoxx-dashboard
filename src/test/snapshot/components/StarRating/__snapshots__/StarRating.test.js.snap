exports[`test StarRating with rating 0 1`] = `
<div
  className="row"
  left="xs">
  <div
    className="cWcebC" />
  <div
    className="cWcebC" />
  <div
    className="cWcebC" />
  <div
    className="cWcebC" />
  <div
    className="cWcebC" />
</div>
`;

exports[`test StarRating with rating 1 1`] = `
<div
  className="row"
  left="xs">
  <div
    className="kmDnAn" />
  <div
    className="cWcebC" />
  <div
    className="cWcebC" />
  <div
    className="cWcebC" />
  <div
    className="cWcebC" />
</div>
`;

exports[`test StarRating with rating 2 1`] = `
<div
  className="row"
  left="xs">
  <div
    className="kmDnAn" />
  <div
    className="kmDnAn" />
  <div
    className="cWcebC" />
  <div
    className="cWcebC" />
  <div
    className="cWcebC" />
</div>
`;

exports[`test StarRating with rating 3 1`] = `
<div
  className="row"
  left="xs">
  <div
    className="kmDnAn" />
  <div
    className="kmDnAn" />
  <div
    className="kmDnAn" />
  <div
    className="cWcebC" />
  <div
    className="cWcebC" />
</div>
`;

exports[`test StarRating with rating 4 1`] = `
<div
  className="row"
  left="xs">
  <div
    className="kmDnAn" />
  <div
    className="kmDnAn" />
  <div
    className="kmDnAn" />
  <div
    className="kmDnAn" />
  <div
    className="cWcebC" />
</div>
`;

exports[`test StarRating with rating 5 1`] = `
<div
  className="row"
  left="xs">
  <div
    className="kmDnAn" />
  <div
    className="kmDnAn" />
  <div
    className="kmDnAn" />
  <div
    className="kmDnAn" />
  <div
    className="kmDnAn" />
</div>
`;
